src/dataModels/JSON.js

```diff
@@ -607,42 +614,20 @@ var JSON = DataModel.extend('dataModels.JSON', {
 
     /**
      * @memberOf dataModels.JSON.prototype
-     * @param {number} y
+     * @param {number} r - Grid row coordinate.
      * @returns {object}
      */
-    getRow: function(y) {
-        var headerRowCount = this.grid.getHeaderRowCount();
-        var topTotals = this.getTopTotals();
-        var hasToptotals = !!topTotals.length;
-        if (y < headerRowCount && !hasToptotals) {
+    getRow: function(r) {
+        var headerRowCount = this.grid.getHeaderRowCount(),
+            topTotals = this.getTopTotals(),
+            hasToptotals = !!topTotals.length,
+            y = r - headerRowCount;
 
-            return topTotals[y - (headerRowCount - topTotals.length)];
+        if (r < headerRowCount && !hasToptotals) {
+            return topTotals[r - (headerRowCount - topTotals.length)];
         }
-        return this.dataSource.getRow(y - headerRowCount);
-    },
-
-    buildRow: function(y) {
-        return this.deprecated('buildRow', 'dataSource', '1.0.8');
-    },
-
-    /**
-     * @memberOf dataModels.JSON.prototype
-     * @param {number} y
-     * @returns {object}
-     */
-    getComputedRow: function(y) {
-        var rcf = this.getRowContextFunction([y]);
```

Comment:
Did you test to confirm that aggregations/grouping was not broken by doing this?